retrofit/src/main/java/retrofit/Call.java

```diff
@@ -0,0 +1,72 @@
+package retrofit;
+
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicReference;
+
+public final class Call<T> {
+  private final Execution execution;
+  private final Executor callbackExecutor;
+
+  private volatile com.squareup.okhttp.Call rawCall;
+  private boolean executed; // Guarded by this.
+
+  Call(Execution execution, Executor callbackExecutor) {
+    this.execution = execution;
+    this.callbackExecutor = callbackExecutor;
+  }
+
+  public void enqueue(final Callback<T> callback) {
+    synchronized (this) {
+      if (executed) throw new IllegalStateException("Already Executed");
+      executed = true;
+    }
+    rawCall = execution.execute(new ExecutionCallback() {
```

Comment:
I'm surprised it's called `execute` and takes a callback. In OkHttp `enqueue` takes a callback; `execute` is synchronous and returns a value.